services/src/northbound/src/main/java/org/openkilda/northbound/service/impl/FlowServiceImpl.java

```diff
@@ -520,30 +525,30 @@ public String toString() {
 
                     rule = new SimpleSwitchRule();
                     rule.switchId = inNode.getSwitchId();
-                    rule.inPort = ""+inNode.getPortNo();
-                    rule.cookie = ""+inNode.getCookie();
-                    if (rule.cookie == null || rule.cookie.length() == 0 || rule.cookie.equals("null"))
-                        rule.cookie = ""+flow.getCookie();
-                    rule.inVlan = ""+flow.getTransitVlan();
-                    rule.outVlan = ""+flow.getTransitVlan();
-                    rule.outPort = ""+outNode.getPortNo();
+                    rule.inPort = inNode.getPortNo();
+                    rule.cookie = Optional.ofNullable(inNode.getCookie())
+                            .filter(cookie -> !cookie.equals(NumberUtils.LONG_ZERO))
```

Comment:
NumberUtils.LONG_ZERO is very generic and not related to the use cases. Can we have ZERO_COOKIE constant?